source/beamline/AMDetector.cpp

```diff
@@ -428,28 +459,34 @@ bool AMDetector::clear(){
 	return clearImplementation();
 }
 
-void AMDetector::setAsDarkCurrentMeasurementValue(){
+void AMDetector::setLastMeasurementAsDarkCurrent()
+{
+	if (canDoDarkCurrentCorrection()) {
+		double newTime = acquisitionTime();
+		double newValue = double(singleReading()) / newTime;
+
+		updateDarkCurrentTime(newTime);
+		updateDarkCurrentValue(newValue);
+		updateDarkCurrentValidState(true);
```

Comment:
Just for naming's sake, I'd change these from update methods to set methods.  This is particularly true for ```updateDarkCurrentValidState()``` as it reads "Should I check if I'm valid? Yes." instead of "You are now valid."